src/client/debugger/jupyter/debugger.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+import { debug, NotebookDocument, DebugSession, DebugSessionOptions, DebugConfiguration } from 'vscode';
+
+export class Debugger {
+    private resolveFunc?: (value: DebugSession) => void;
+    private rejectFunc?: (reason?: Error) => void;
+
+    readonly session: Promise<DebugSession>;
+
+    constructor(
+        public readonly document: NotebookDocument,
+        public readonly config: DebugConfiguration,
+        options?: DebugSessionOptions
+    ) {
+        this.session = new Promise<DebugSession>((resolve, reject) => {
```

Comment:
You might try using a deferred object for this idea of remembering the promise resolve/reject. 

If you create a deferred object, you can implement resolve/reject with it.